src/mongo-read-stream.ts

```diff
@@ -173,11 +167,54 @@ export class MongoReadStream<
 				item = await this.cursor.next();
 			}
 			if (item) {
-				this.lastId = item._id;
+				this.lastItem = { ...LodashReplacerUtils.PICK_PROPERTIES(item, Object.keys(this.sort)) };
+				if (this.limit) this.limit -= 1;
 			}
 			this.push(item);
 		} catch (e) {
 			this.emit('error', e);
 		}
 	}
+
+	private updateQueryWithSortParams(): void {
+		this._query.$and = this._query.$and ?? [];
+		if (LodashReplacerUtils.IS_OBJECT_EMPTY(this.sort)) return;
```

Comment:
Could you add a test with empty sort ? I think we should always sort at least by _id instead of nothing, no ?